dandi/metadata/util.py

```diff
@@ -583,6 +584,29 @@ def extract_digest(metadata: dict) -> dict[models.DigestType, str] | None:
         return None
 
 
+def extract_related_resource(metadata: dict) -> list[models.Resource] | None:
+    pubs = metadata.get("related_publications")
+    if not isinstance(pubs, (list, tuple)):
+        return None
+    related = []
+    for v in pubs:
+        if not isinstance(v, str):
+            continue
+        try:
+            u = URL(v)
+        except ValueError:
+            continue
+        if u.scheme not in ("http", "https") or u.host != "doi.org":
+            continue
+        related.append(
+            models.Resource(
+                identifier=v,
+                relation=models.RelationType.IsDescribedBy,
```

Comment:
I think this one is is too specific to assume that any reference is the description of the data we find in the file.
What we can say only that the file references that publication, and hence I would better go with 

```suggestion
                relation=models.RelationType.References,
```

as the default.  WDYT @bendichter @satra ?

Reply:
I see your point that we cannot necessarily infer the relationship between the data and the paper.

The [DataCite definition for "References"](https://datacite-metadata-schema.readthedocs.io/en/4.5/appendices/appendix-1/relationType/#references) is "indicates B is used as a source of information for A." Here, this would mean that the paper is used as a source of information for the Dandiset. I don't know if this really fits here.

I have been wondering what the best way to associate papers with datasets using these relations. What are the different types of ways a paper and a dataset can be associated? So far, I have been using [isDescribedBy](https://datacite-metadata-schema.readthedocs.io/en/4.5/appendices/appendix-1/relationType/#isdescribedby) (indicates that A describes B) for everything, assuming that the publication is the primary publication in which the data is introduced and described. Another type of relationship might be that a paper reuses a dataset. In this case, [IsSupplementTo](https://datacite-metadata-schema.readthedocs.io/en/4.5/appendices/appendix-1/relationType/#issupplementto) (indicates that A is a supplement to B ) might be a better choice, since it applies to both. Are there other relationship types that describe how a paper and dandiset might be related? What are the different use-cases here?

This is challenging in part because there is currently no aspect of the dandi schema that allows us to describe what the resource is (see issue [here](https://github.com/dandi/dandi-schema/issues/231)), so that needs to be inferred at least somewhat by the relation type.

Reply:
@bendichter Now that dandi/dandi-schema#231 has been resolved, how can we move this PR forwards?

Reply:
Yes, exactly. My preference would be to change to schema to make this optional. Now that we have a resource type I am much less reliant on this particular field, though I do appreciate the caveat that optional fields are very rarely populated. I would also be happy with defaulting to IsDescribedBy or IsSupplementTo.

Reply:
ok, to expedite, let's proceed with IsDescribedBy and see where it takes us ;-)